[CCI-1818] Add documentation for CCM EBS shared cost allocation (#20994)
DataDog/documentation: content/en/cloud_cost_management/container_cost_allocation.md

### Kubernetes

#### Compute

For Kubernetes compute allocation, a Kubernetes node is joined with its associated EC2 instance costs. The node's cluster name and all node tags are added to the entire EC2 compute cost for the node. This allows you to associate cluster-level dimensions with the cost of the instance, without considering the pods scheduled to the node.

Next, Datadog looks at all of the pods running on that node for the day. The cost of the node is allocated to the pod based on the resources it has used and the length of time it ran. This calculated cost is enriched with all of the pod's tags.

**Note**: Only _tags_ from pods and nodes are added to cost metrics. To include labels, enable labels as tags for [nodes][7] and [pods][8].

#### Persistent volume storage

For Kubernetes Persistent Volume storage allocation, Persistent Volumes (PV), Persistent Volume Claims (PVC), nodes, and pods are joined with their associated EBS volume costs. All associated PV, PVC, node, and pod tags are added to the EBS volume cost line items.

> Next, Datadog looks at all of the pods that claimed the volume on that day. The cost of the volume is allocated to the pod based on the resources it has used and the length of time it ran. These resources

Reviewer: ```suggestion
Next, Datadog looks at all of the pods that claimed the volume on that day. The cost of the volume is allocated to a pod based on the resources it used and the length of time it ran. These resources
```

> include the provisioned capacity for Storage, IOPS, and Throughput. This allocated cost is enriched with all of the pod's tags.

Reviewer: ```suggestion
include the provisioned capacity for storage, IOPS, and throughput. This allocated cost is enriched with all of the pod's tags.
```

### Everything else

> Cost other than EC2/EBS, computed for instances hosting Kubernetes pods or ECS tasks, is given the same value and tags as the source metric, `aws.cost.amortized`.

Reviewer: ```suggestion
All other costs (costs that are not related to EC2 or EBS, and are not computed for instances hosting Kubernetes pods or ECS tasks) are given the same value and tags as the source metric, `aws.cost.amortized`.
```

@kaankoseler I'm trying to reword this sentence to make it more clear. Can you please check that my change doesn't change the intended meaning of the sentence?

Author: Hi @cswatt thank you for the review.

This is a tough sentence to work with, I think. The revision does seem to lose the meaning somewhat.  
The idea that we are trying to convey is that this subset of costs is what's being targeted with the feature:
`EC2/EBS costs for EC2 instances hosting Kubernetes pods or ECS tasks`

Any cost that does not fall under that umbrella is considered `Everything else` and left as is with the source metric. 
So we somehow need to keep that sentence above as one coherent whole, IMO.

Reviewer 2: As the list of allocated cost types is growing, this sentence might just not be scalable 😆 I wonder if we even need to re-list all the costs above. One option is just drop the explanation:
"All other costs are given the same value and tags as the source metric, `aws.cost.amortized`."

Or, if we want an explanation, to split it up into multiple sentences:
"All other costs are given the same value and tags as the source metric, `aws.cost.amortized`. This includes costs for all other AWS services, as well as EC2 and EBS costs not associated with container orchestration."

Author: I would prefer the former, personally. It should be pretty obvious from the sections and the preceding text of the document

Author: @cswatt what do you think about just keeping this as:
`"All other costs are given the same value and tags as the source metric, aws.cost.amortized."`

Reviewer: How about "The information on this page pertains to EC2/EBS costs for EC2 instances hosting Kubernetes pods or ECS tasks. All other costs are given the same value and tags as the source metric, `aws.cost.amortized`." —does that work?

Author: Yes, that works! cc @enemyghost

Author: I talked to @enemyghost offline, can we slightly modify to say `... pertains to EC2 Compute and EBS costs...`?

> Cost other than EC2/EBS, computed for instances hosting Kubernetes pods or ECS tasks, is given the same value and tags as the source metric, `aws.cost.amortized`.

Reviewer: ```suggestion
The information on this page pertains to EC2 compute and EBS costs for EC2 instances hosting Kubernetes pods or ECS tasks. All other costs are given the same value and tags as the source metric, `aws.cost.amortized`.
```

## Understanding spend

Using the `allocated_spend_type` tag, you can visualize the spend category associated with your cost at the Kubernetes node, ECS host, EBS volume, or cluster level. Costs are allocated into three spend types:

### Compute

- Usage: Cost of memory and cpu being used or reserved by workloads.
- Workload idle: Cost of memory and cpu that is being reserved and allocated but not used by workloads.
- Cluster idle: Cost of memory and cpu not reserved by workloads in a cluster.

### Persistent volumes

> - Usage: Cost of provisioned IOPS, Throughput, or Storage being used by workloads. For Storage, this is based on the maximum amount of volume storage used that day. For IOPS and Throughput, this is based on the average amount used that day.

Reviewer: ```suggestion
- Usage: Cost of provisioned IOPS, throughput, or storage being used by workloads. Storage cost is based on the maximum amount of volume storage used that day. IOPS and throughput costs are based on the average amount used that day.
```

> - Workload idle: Cost of provisioned IOPS, Throughput, or Storage not being used by workloads. For Storage, this is based on the maximum amount of storage used that day. For IOPS and Throughput, this is based on the average amount used that day.

Reviewer: ```suggestion
- Workload idle: Cost of provisioned IOPS, throughput, or storage not being used by workloads. Storage cost is based on the maximum amount of storage used that day. IOPS and throughput costs are based on the average amount used that day.
```

> - Cluster idle: Cost of provisioned IOPS, Throughput, or Storage for volumes not claimed by any pods that day.

Reviewer: ```suggestion
- Cluster idle: Cost of provisioned IOPS, throughput, or storage for volumes not claimed by any pods that day.
```

| `allocated_resource:cpu`                   | Cost of CPU resources. *Only available for `.shared.resources.allocated` metrics.*                                            |
| `allocated_resource:memory`                | Cost of Memory resources. *Only available for `.shared.resources.allocated` metrics.*                                         |
| `allocated_resource:managed_service_fee`   | Cost of cloud provider managed service fees. *Only available for `.shared.resources.allocated` metrics.*           |

> | `allocated_resource:persistent_volume`     | Cost of non-local EBS Volumes used as Persistent Volumes in Kubernetes. *Only available for `.shared.resources.allocated` metrics.*           |

Reviewer: ```suggestion
| `allocated_resource:persistent_volume`     | Cost of non-local EBS volumes used as Persistent Volumes in Kubernetes. *Only available for `.shared.resources.allocated` metrics.*           |
```

### Persistent volumes

In addition to Kubernetes pod and Kubernetes node tags, the following out-of-the-box tags are applied to cost metrics.

| Out-of-the-box tag                      | Description                                                                                                                                 |
| ---                                     |---------------------------------------------------------------------------------------------------------------------------------------------|
| `persistent_volume_reclaim_policy`      | The Kubernetes Reclaim Policy on the Persistent Volume.                                                                                     |
| `storage_class_name`                    | The Kubernetes Storage Class used to instantiate the Persistent Volume.                                                                     |
| `volume_mode`                           | The Volume Mode of the Persistent Volume.                                                                                                   |

> | `ebs_volume_type`                       | The type of the AWS EBS volume. *Can be `gp3`, `gp2`, and more.*                                                                            |

Reviewer: ```suggestion
| `ebs_volume_type`                       | The type of the AWS EBS volume. Can be `gp3`, `gp2`, or others.                                                                            |
```
